fix(openai): Remove automatic summary from reasoning_effort transformation

Chesars opened the pull request:
## Title

fix: Remove automatic summary from reasoning_effort transformation
  
## Relevant issues

Fixes #16032
 
## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [ ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🐛 Bug Fix
# Changes

  ## Summary
  The `_map_reasoning_effort()` function was automatically adding `reasoning.summary` field when users specified `reasoning_effort` parameter,  causing **400 errors** for users with unverified OpenAI organizations:

>   Your organization must be verified to generate reasoning summaries.
>   Please go to: https://platform.openai.com/settings/organization/general

  ## Details
  According to [OpenAI's official documentation](https://platform.openai.com/docs/guides/reasoning#reasoning-summaries), the `summary` field is **opt-in** and requires organization verification:

  > "Reasoning summary output [...] **will not be included unless you explicitly opt in** to including reasoning summaries."
  >
  > "Before using summarizers with our latest reasoning models, **you may need to complete organization verification** to ensure safe deployment."

  The previous implementation automatically adds `summary="detailed"` or `summary="auto"` to all `reasoning_effort` requests.

  ### Solution
  Remove the automatic inclusion of `summary` field from all `reasoning_effort` levels:

  **Before:**
  ```python
  def _map_reasoning_effort(self, reasoning_effort: str) -> Optional[Reasoning]:
      if reasoning_effort == "high":
          return Reasoning(effort="high", summary="detailed")  # ❌ Auto-adds summary
      elif reasoning_effort == "medium":
          return Reasoning(effort="medium", summary="auto")    # Same here
      # ...
```
  After:
```python
  def _map_reasoning_effort(self, reasoning_effort: str) -> Optional[Reasoning]:
      if reasoning_effort == "high":
          return Reasoning(effort="high")  
      elif reasoning_effort == "medium":
          return Reasoning(effort="medium")   
      # ...
```
  Users who want reasoning summaries can explicitly opt-in:
  ### Explicit opt-in (requires verified org)
```python
  litellm.completion(
      model="gpt-5-mini",
      messages=[...],
      reasoning={"effort": "high", "summary": "auto"}
  )
```

  - ✅ Fixes #16032 - Users with unverified orgs can now use reasoning_effort
  - ✅ Works for all organizations (verified and unverified)
  - ✅ Maintains backward compatibility - Users passing reasoning object directly are unaffected
  - ✅ Follows OpenAI's recommended opt-in approach
  - ✅ All existing tests pass (4/4 tests in transformation suite)

  Files Changed

  - litellm/completion_extras/litellm_responses_transformation/transformation.py

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Stops auto-adding reasoning summaries when mapping `reasoning_effort` and allows passing a full `reasoning` dict through.
> 
> - **Reasoning mapping**:
>   - Update `_map_reasoning_effort` to accept `Union[str, Dict[str, Any]]`.
>   - If a dict is provided, pass through via `Reasoning(**reasoning_effort)`.
>   - If a string is provided (`high`/`medium`/`low`/`minimal`), return `Reasoning(effort=...)` without setting `summary`.
> - **Behavioral change**:
>   - Removes automatic inclusion of `reasoning.summary`, preventing errors for unverified orgs and enabling explicit opt-in.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 40a7158ffc4f5609f77ecfd6ba5c25da9359578f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Chesars committed:
Fix: Remove automatic summary field from reasoning_effort transformation

Problem:
The _map_reasoning_effort() function was automatically adding
reasoning.summary field when users specified reasoning_effort parameter,
causing 400 errors for users with unverified OpenAI organizations.

Root Cause:
According to OpenAI's official documentation, the summary field is opt-in
and requires organization verification:

"Reasoning summary output [...] will not be included unless you explicitly
opt in to including reasoning summaries."

"Before using summarizers with our latest reasoning models, you may need
to complete organization verification"

Source: https://platform.openai.com/docs/guides/reasoning#reasoning-summaries

Solution:
Remove the automatic inclusion of summary field from all reasoning_effort
levels (high, medium, low, minimal). Users who want reasoning summaries
can explicitly pass reasoning={"effort": "high", "summary": "auto"} in
their requests.

Impact:
- Fixes #16032
- Works for all organizations (verified and unverified)
- Maintains backward compatibility for users passing reasoning object directly
- Follows OpenAI's recommended opt-in approach

Testing:
- All existing tests pass (4/4 tests in transformation suite)
- Manual verification confirms only effort field is included

Chesars committed:
test: Fix MockResponse missing headers attribute in test_openai_responses_api

The MockResponse class was missing the 'headers' attribute which caused
APIConnectionError when processing the mock response. Added headers={}
to fix the test.

krrishdholakia commented on litellm/completion_extras/litellm_responses_transformation/transformation.py:
can we allow the user to pass in reasoning_effort={"effort": "high", "summary": "detailed"}, to preserve this ability? @Chesars

Chesars replied:
Hi @krrishdholakia. Sure, make sense. Let's add that.

Chesars committed:
feat: Add dict support to reasoning_effort parameter

Allow users to pass reasoning_effort as either:
- String: reasoning_effort="high" (no summary, safe default)
- Dict: reasoning_effort={"effort": "high", "summary": "detailed"} (opt-in)

This preserves backward compatibility while giving users flexibility
to explicitly opt-in to the summary field when needed (for verified
OpenAI organizations).

Chesars committed:
Merge branch 'BerriAI:main' into fix/reasoning-effort-remove-summary-field

Commit:
Merge branch 'main' into fix/reasoning-effort-remove-summary-field

jasonpnnl commented:
This breaks using the completions API to Responses API bridge with OpenWebUI as there is no longer a way to get reasoning summaries.
